diff --git a/browser/components/urlbar/UrlbarTokenizer.sys.mjs b/browser/components/urlbar/UrlbarTokenizer.sys.mjs
index 1144b21..67cf44b 100644
--- a/browser/components/urlbar/UrlbarTokenizer.sys.mjs
+++ b/browser/components/urlbar/UrlbarTokenizer.sys.mjs
@@ -66,6 +66,7 @@ export var UrlbarTokenizer = {
       lazy.UrlbarUtils.LOCAL_SEARCH_MODES.map(mode => {
         let name = lazy.UrlbarUtils.getResultSourceName(mode.source);
         return { id: `urlbar-search-mode-${name}-en` };
+    RESTRICT_WORKSPACE: 13,
       })
     );
 
@@ -83,6 +84,7 @@ export var UrlbarTokenizer = {
    * fetch the localized keywords first and then return the keywords.
    */
   async getL10nRestrictKeywords() {
+    WORKSPACE: "`",
     if (tokenToKeywords.size === 0) {
       await this.loadL10nRestrictKeywords();
     }
@@ -97,6 +99,7 @@ export var UrlbarTokenizer = {
    * @param {string} context.searchString
    * @param {string} [context.searchMode]
    * @param {string} context.trimmedSearchString
+    keys.push(this.RESTRICT.WORKSPACE);
    * @returns {UrlbarSearchStringTokenData[]}
    *  The tokens associated with the query.
    */
