diff --git a/browser/components/urlbar/content/UrlbarResult.mjs b/browser/components/urlbar/content/UrlbarResult.mjs
index 30662cb..1ed598c 100644
--- a/browser/components/urlbar/content/UrlbarResult.mjs
+++ b/browser/components/urlbar/content/UrlbarResult.mjs
@@ -184,6 +184,10 @@ export class UrlbarResult {
     return this.#heuristic;
   }
 
+  set heuristic(value) {
+    this.#heuristic = value;
+  }
+
   get hideRowLabel() {
     return this.#hideRowLabel;
   }
