diff --git a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
index d2e2415..7a08011 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
+++ b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
@@ -914,6 +914,7 @@ export class Query {
     if (
       result.heuristic &&
       this.context.searchMode &&
+      !(this.context.searchMode.source === lazy.UrlbarUtils.RESULT_SOURCE.ZEN_ACTIONS && result.payload?.zenAction) &&
       (!this.context.trimmedSearchString ||
         (!this.context.searchMode.engineName && !result.autofill))
     ) {
@@ -1044,6 +1045,7 @@ function updateSourcesIfEmpty(context) {
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_TITLE,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_URL,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_ACTION,
+            lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE,
           ].includes(t.type)
         );
 
@@ -1103,6 +1105,14 @@ function updateSourcesIfEmpty(context) {
           acceptedSources.push(source);
         }
         break;
+      case lazy.UrlbarUtils.RESULT_SOURCE.WORKSPACES:
+        if (
+          restrictTokenType === lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
       case lazy.UrlbarUtils.RESULT_SOURCE.OTHER_LOCAL:
       case lazy.UrlbarUtils.RESULT_SOURCE.ADDON:
       default:
