diff --git a/dom/chrome-webidl/MediaController.webidl b/dom/chrome-webidl/MediaController.webidl
index 790bc66..b9be1e3 100644
--- a/dom/chrome-webidl/MediaController.webidl
+++ b/dom/chrome-webidl/MediaController.webidl
@@ -23,6 +23,12 @@ enum MediaControlKey {
   "setvolume",
 };
 
+dictionary MediaControllerPositionState {
+  required double duration;
+  required double playbackRate;
+  required double position;
+};
+
 /**
  * MediaController is used to control media playback for a tab, and each tab
  * would only have one media controller, which can be accessed from the
@@ -42,10 +48,14 @@ interface MediaController : EventTarget {
   // management strategies based on the kind of audio the tab is playing.
   [BinaryName="GetEffectiveAudioSessionType"]
   readonly attribute AudioSessionType effectiveAudioSessionType;
+  readonly attribute boolean isBeingUsedInPIPModeOrFullscreen;
 
   [Throws]
   MediaMetadataInit getMetadata();
 
+  [Throws]
+  MediaControllerPositionState getPositionState();
+
   [Frozen, Cached, Pure]
   readonly attribute sequence<MediaControlKey> supportedKeys;
 
